compiler/packages/snap/src/compiler.ts

```diff
@@ -174,9 +174,20 @@ function makePluginOptions(
       .filter(s => s.length > 0);
   }
 
-  let inferEffectDependencies = false;
+  let inferEffectDependencies = null;
   if (firstLine.includes('@inferEffectDependencies')) {
-    inferEffectDependencies = true;
+    inferEffectDependencies = [
+      {
+        module: 'react',
+        imported: 'useEffect',
+        numRequiredArgs: 1,
+      },
+      {
+        module: 'react',
+        imported: 'useSpecialEffect',
+        numRequiredArgs: 2,
```

Comment:
Nit: could this import be from module: `shared-runtime`, and we add the relevant export to `shared-runtime.ts`?

It would be great to be able to evaluate all fixture tests, and adding an unknown import from `react` would be a little sketchy anyways

Reply:
Instead of test-fixture specific defaults, could we also add a default value to [`testComplexConfigDefaults`](https://github.com/facebook/react/blob/daa506e8b1692aa9d3d9f620fa034993c5341f6d/compiler/packages/babel-plugin-react-compiler/src/HIR/Environment.ts#L609-L649) so that playground users can use `@inferEffectDependencies`?